tutorials/pretraining-Vietnamese-data-curation/pretraining-Vietnamese-data-curation.ipynb

```diff
@@ -0,0 +1,655 @@
+    "    labeled_dataset = label_quality(raw_dataset)\n",
+    "    labeled_samples = labeled_dataset.df.sample(frac=num_samples / len(labeled_dataset.df))\n",
+    "    return labeled_samples[\"text\"].compute().values.tolist()\n",
+    "\n",
+    "# Prepare training data\n",
+    "low_quality_samples = create_samples(lq_samples_path, \"__label__lq\", 100000)\n",
+    "high_quality_samples = create_samples(hq_samples_path, \"__label__hq\", 100000)\n",
+    "train_samples = low_quality_samples + high_quality_samples\n",
+    "random.shuffle(train_samples)\n",
+    "\n",
+    "# Save training data to a file\n",
+    "train_file = \"./cf_model_fasttext.train\"\n",
+    "with open(train_file, \"w\") as f:\n",
+    "    for sample in train_samples:\n",
+    "        f.write(sample + \"\\n\")\n",
+    "\n",
+    "# Train the FastText classifier\n",
+    "model = fasttext.train_supervised(input=train_file, lr=0.01, dim=100, epoch=5, wordNgrams=2)\n",
+    "model_path = \"./cf_model_fasttext_model.bin\"\n",
+    "model.save_model(model_path)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Classify and filter the dataset**\n",
+    "\n",
+    "Once trained, the classifier is used to filter the dataset, categorizing documents into high and low quality based on the learned distinctions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nemo_curator.filters import FastTextQualityFilter\n",
+    "from nemo_curator import ScoreFilter\n",
+    "\n",
+    "# Define paths and load the dataset\n",
+    "CF_input_data_dir = HF_output_path\n",
+    "CF_output_path = os.path.join(data_dir, 'classifier_filtering/output')\n",
+    "target_dataset = DocumentDataset.read_parquet(CF_input_data_dir, 'parquet')\n",
+    "\n",
+    "# Set up the filtering pipeline\n",
+    "filter_pipeline = ScoreFilter(FastTextQualityFilter(model_path), score_field=\"quality_score\", score_type=float)\n",
+    "filtered_dataset = filter_pipeline(target_dataset)\n",
+    "\n",
+    "# Save the filtered dataset\n",
+    "write_to_disk(filtered_dataset.df, output_file_dir=CF_output_path, write_to_filename=True, output_type='parquet')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Close the CPU Dask Cluster"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.cluster.close()\n",
+    "client.shutdown()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We have completed the notebook! For other techniques such as Fuzzy Deduplication or PII redaction, you can go to [NeMo Curator example scripts](https://github.com/NVIDIA/NeMo-Curator/tree/main/examples)."
+   ]
+  }
+ ],
+ "metadata": {
+  "language_info": {
+   "name": "python"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2
+}
```

Comment:
Can you please run the full notebook from top to bottom, so we can see all the outputs?